icu4c/source/common/uniset.cpp

```diff
@@ -981,6 +981,7 @@ void UnicodeSet::_add(const UnicodeString& s) {
     }
     strings->sortedInsert(t, compareUnicodeString, ec);
     if (U_FAILURE(ec)) {
+        delete t;
```

Comment:
I think this is incorrect, and that the original code is OK.

The `strings` UVector in UnicodeSet is set up with a deleter function. see uniset.cpp:1610.
Which means that `strings->sortedInsert(t, ...)` takes ownership the element being inserted, t. This is true always, failure or no. In the event of a failure, the `UVector` automatically deletes the item that it could not insert. See uvector.cpp:463

Reply:
Good catch! Thanks Andy!